ccp/src/cu/proving_thread/sync/to_state_operating_message.rs

```diff
@@ -62,8 +66,15 @@ pub(crate) enum StateOperatingControlMessage {
     CleanProofCache {
         notifier: oneshot::Sender<tokio::io::Result<()>>,
     },
-    GetProofs {
-        proof_idx: ProofIdx,
-        notifier: oneshot::Sender<tokio::io::Result<Vec<CCProof>>>,
+    GetAllProofs {
+        last_known_proofs: HashMap<CUID, ProofIdx>,
+        limit: usize,
+        notifier: oneshot::Sender<tokio::io::Result<Vec<BatchResponse>>>,
+    },
+    GetBatchProofs {
+        reqs: HashMap<CUID, BatchRequest>,
```

Comment:
Let's serialise CUID as hex. 

Re impl: if serde poses complications for HashMaps, try to use newtype for CUID or use `Vec<CUID, ...` or newtype for `HashMap<CUID, ...`